test/test_utils.py

```diff
@@ -108,6 +118,9 @@ def assert_expected_wrapper(actual, expected):
         elif isinstance(_actual, tuple):
             # outputs are from multiple layers: (Tensor, Tensor, ...)
             assert_expected_wrapper(tuple_to_dict(_actual), tuple_to_dict(_expected))
+        elif is_named_tuple(_actual):
+            # output is another named tuple instance
+            assert_expected_wrapper(_actual, _expected)
```

Comment:
very basic q, why cant people write their loops instead of trying to generalize this method for every datatype?